### PR TITLE
Component/select

### PR DESCRIPTION
**Pull request by admtomas**
Just refactored a code as Global styles did not worked. Previously we could not select any options from dropdown list. Now it is sorted.

Please check it as other work depend on it and we ned to merge it asap.

### CODE REVIEW

File: src/components/Select/Select.jsx [hunk 1]
```diff
@@ -1,35 +1,40 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { SelectWrapper } from './Select.styles';
```
Comment (catherinehoward) [inline src/components/Select/Select.jsx:3]: Need to add this back in rename to SelectInputWrapper

Comment (catherinehoward) [inline src/components/Select/Select.jsx:3]: Need to add this back in and rename to SelectInputWrapper:

import { SelectInputWrapper } from './Select.styles';

File: src/components/Select/Select.jsx [hunk 2]
```diff
@@ -1,35 +1,40 @@
...
+import { SelectContainer, LabelTag, SelectTag } from './Select.styles';
+import InputWrapper from '../../theme/inputs.styles';
```
Comment (catherinehoward) [inline src/components/Select/Select.jsx]: amend to import { SelectInputWrapper } from './Select.styles';

You don't need to use LabelTag or SelectTag

Delete import InputWrapper from '../../theme/inputs.styles';
InputWrapper needs to be be imported in Select.styles.js . (See TextInput and TextArea components)

File: src/components/Select/Select.jsx [hunk 3]
```diff
@@ -1,35 +1,40 @@
...
-      <select name={name} id={name} onBlur={onBlur} value={defaultValue}>
-        {options.map(({ optionValue, optionLabel }) => (
-          <option
-            key={optionLabel}
-            className="select--option"
-            value={optionValue}
-          >
-            {optionLabel}
+    <InputWrapper>
+      <LabelTag htmlFor={name}>{label}</LabelTag>
```
Comment (catherinehoward) [inline src/components/Select/Select.jsx]: Change to:
`<SelectInputWrapper>`
`<label htmlFor={name}>{label}</label>`

File: src/components/Select/Select.jsx [hunk 4]
```diff
@@ -1,35 +1,40 @@
...
+      <SelectContainer>
```
Comment (catherinehoward) [inline src/components/Select/Select.jsx]: No need to use `<SelectContainer>` so remove.

File: src/components/Select/Select.jsx [hunk 5]
```diff
@@ -1,35 +1,40 @@
...
+        <SelectTag name={name} id={name} onBlur={onBlur}>
+          <option value="" selected disabled hidden>
```
Comment (catherinehoward) [inline src/components/Select/Select.jsx]: Change to `<select name={name} id={name} onBlur={onBlur}>`

File: src/components/Select/Select.stories.jsx [hunk 6]
```diff
@@ -10,12 +10,11 @@ export const Fruit = () => (
   <Select
     name="Fruit"
     label="Choose a fruit"
-    defaultValue="Fruits"
+    placeholder="I am default Value"
```
Comment (catherinehoward) [inline src/components/Select/Select.stories.jsx]: Change placeholder to "Select an option..."

File: src/components/Select/Select.stories.jsx [hunk 7]
```diff
@@ -29,10 +28,10 @@ export const Example = () => (
     label="Example label"
     defaultValue="Example default value"
```
Comment (catherinehoward) [inline src/components/Select/Select.stories.jsx]: Change defaultValue to placeholder

File: src/components/Select/Select.styles.js [hunk 8]
```diff
@@ -1,14 +1,31 @@
 import styled from 'styled-components';
```
Comment (catherinehoward) [inline src/components/Select/Select.styles.js:3]: Add
`import InputWrapper from '../../theme/inputs.styles';`

File: src/components/Select/Select.styles.js [hunk 9]
```diff
@@ -1,14 +1,31 @@
...
+  appearance: none;
+  -moz-appearance: none;
+  -webkit-appearance: none;
+`;
+
+export const LabelTag = styled.label`
+  font-weight: bold;
+  color: var(--darkblue);
+`;
+
```
Comment (catherinehoward) [inline src/components/Select/Select.styles.js]: Delete this code - not needed. The SelectTag styling is duplicating select styling below.
LableTag styling is already defined in typography.styles.js

File: src/components/Select/Select.styles.js [hunk 10]
```diff
@@ -17,8 +34,8 @@ export const SelectWrapper = styled.div`
     content: ' ';
     display: block;
     position: absolute;
-    top: 5.2rem;
-    right: var(--medium-space);
+    top: 2.5rem;
```
Comment (catherinehoward) [inline src/components/Select/Select.styles.js]: Need to change this to 5rem or try 5.2rem. Whichever looks best.

File: src/components/Select/Select.jsx (hunk 1 above)
Comment (admtomas) [reply src/components/Select/Select.jsx:3]: done it

File: src/components/Select/Select.styles.js (hunk 10 above)
Comment (admtomas) [reply src/components/Select/Select.styles.js]: All done as advides please have a look.

File: src/components/Select/Select.styles.js [hunk 11]
```diff
@@ -1,14 +1,16 @@
...
+
+  label {
+    font-weight: bold;
+  }
 
   select {
-    font-size: 1.6rem;
-    cursor: pointer;
-    color: var(--darkgrey);
+    appearance: none;
```
Comment (catherinehoward) [inline src/components/Select/Select.styles.js:9]: Think we still need to include the font-size: 1.6rem so it's the same as TextInput and TextArea components. It's better for accessibility. It should be 1.6rem by default. I don't think the placeholder styling in /theme/inputs.styles is working. I'll create a seperate issue.

File: src/components/Select/Select.styles.js [hunk 12]
```diff
@@ -1,14 +1,16 @@
...
-export const SelectWrapper = styled.div`
+export const SelectInputWrapper = styled(InputWrapper)`
   display: flex;
-  flex-flow: column;
-  position: relative;
+  flex-direction: column;
+
+  label {
+    font-weight: bold;
+  }
```
Comment (catherinehoward) [inline src/components/Select/Select.styles.js]: The label for TextInput and TextArea is font-weight:400.  No need to include label styling as it should be in the default typography.styles

File: src/components/Select/Select.styles.js (hunk 10 above)
Comment (catherinehoward) [reply src/components/Select/Select.styles.js]: Looking good @admtomas. Just a couple of minor things on the Select.styles.js

File: src/components/Select/Select.styles.js (hunk 11 above)
Comment (admtomas) [reply src/components/Select/Select.styles.js:9]: `<select>` doesnt have placeholder attribute and as it doesnt exist it cannot be styled. It require workaround.

Regarding bold font , i removed that style.

Comment (admtomas) [reply src/components/Select/Select.styles.js:9]: I found some workaround for color of placeholder using `{ useState }`


const [color, setColor] = useState(`var(--darkgrey)`);

  function changeColor() {
    setColor(`var(--darkblue)`);
  }

then insite `<select>` tag i added:
`...
style={{ color }}
onChange={changeColor}
...`

File: src/components/Select/Select.styles.js (hunk 12 above)
Comment (admtomas) [reply src/components/Select/Select.styles.js]: code removed

**Review by catherinehoward — APPROVED** [review]

**Comment by catherinehoward** [comment]
All looks fine.

**Pull request merged**